lib/api/core.dart

```diff
@@ -214,9 +215,39 @@ Map<String, String> authHeader({required String email, required String apiKey})
 }
 
 Map<String, String> userAgentHeader() {
+  final deviceInfo = ZulipBinding.instance.deviceInfo;
+  final PackageInfo(
+    version: appVersion,
+    buildNumber: appBuildNumber) = ZulipBinding.instance.packageInfo;
+
+  final (osName, osVariant) = switch (deviceInfo) {
+    AndroidDeviceInfo(
+      :var sdkInt)        => ('Android', '$sdkInt'), // "34"
+    IosDeviceInfo(
+      :var systemVersion) => ('iOS', systemVersion), // "17.4"
+    MacOsDeviceInfo(
+      :var majorVersion,
+      :var minorVersion,
+      :var patchVersion)  => ('macOS', '$majorVersion.$minorVersion.$patchVersion'), // "14.5.0"
+    WindowsDeviceInfo(
+      :var majorVersion,
+      :var minorVersion,
+      :var buildNumber)   => ('Windows', '$majorVersion.$minorVersion $buildNumber'), // "10.0 22631" (actually means Windows 11, 23H2)
+    LinuxDeviceInfo(
+      :var name,
+      :var versionId)     => ('Linux', '$name${versionId != null ? ' $versionId' : ''}'), // "Fedora Linux 40" or "Fedora Linux"
+    _                     => ('Unknown', ''),
+  };
+  final osInfo = '$osName${osVariant.isNotEmpty ? ', $osVariant' : ''}';
+
   return {
-    // TODO(#467) include platform, platform version, and app version
-    'User-Agent': 'ZulipFlutter',
+    // Possible examples:
+    //  'ZulipFlutter/0.0.15+15 (Android, 14)'
+    //  'ZulipFlutter/0.0.15+15 (iOS, 17.4)'
+    //  'ZulipFlutter/0.0.15+15 (macOS, 14.5.0)'
+    //  'ZulipFlutter/0.0.15+15 (Windows, 10.0 22631)'
+    //  'ZulipFlutter/0.0.15+15 (Linux, Fedora Linux 40)'
+    'User-Agent': 'ZulipFlutter/$appVersion+$appBuildNumber ($osInfo)',
```

Comment:
Is this format okay?

Reply:
Good question. Let's discuss that on `#api design`, on this thread:
https://chat.zulip.org/#narrow/stream/378-api-design/topic/zulip-flutter.20user.20agent/near/1712676

So please post your example values there and we can get feedback on them.